cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/dataproc/DataprocClient.java

```diff
@@ -341,6 +341,11 @@ public ClusterOperationMetadata createCluster(String name, String imageVersion,
       GceClusterConfig.Builder clusterConfig = GceClusterConfig.newBuilder()
         .addServiceAccountScopes(DataprocConf.CLOUD_PLATFORM_SCOPE)
         .putAllMetadata(metadata);
+
+      if (conf.getServiceAccount() != null) {
```

Comment:
should we also check for empty string?

Reply:
It is already taken care of by the `DataprocConf.getString` method.